### PR TITLE
feat(hook): Add customize Netease Cookie

### PR DESCRIPTION
**Pull request by 577fkj**

### CODE REVIEW

File: src/hook.js [hunk 1]
```diff
@@ -128,6 +128,19 @@ hook.request.before = (ctx) => {
...
+	if (process.env.NETEASE_COOKIE && url.path.includes('url')) {
+		var cookies = cookieToMap(req.headers.cookie);
+		var new_cookies = cookieToMap(process.env.NETEASE_COOKIE);
+
+		Object.entries(new_cookies).forEach(([key, value]) => {
+			cookies[key] = value;
+		});
+
+		req.headers.cookie = mapToCookie(cookies);
+		console.log('Replace netease cookie');
```
Comment (1715173329) [inline src/hook.js]: Use `logger.debug` instead.

Comment (577fkj) [reply src/hook.js]: I have made the requested changes and replaced it with console.debug.

Comment (1715173329) [reply src/hook.js]: we use pino to handle logs, the function is called `logger.debug` not `console.debug`.

Comment (577fkj) [reply src/hook.js]: Sorry, I have made the changes as requested.

**Comment by 577fkj** [comment]
更高音质貌似未解锁，不知道是哪个地址获取的

**Comment by 1715173329** [comment]
https://github.com/UnblockNeteaseMusic/server/commit/dd92fe629ab4823a5cd8ee272a90624d73151409

想起来当时胡乱写过一个差不多的功能
请求的 api 还挺多，得抓包分析，我比较懒，当时就放弃了😂

**Comment by 577fkj** [comment]
看了几个关于歌曲信息的包没看到更高音质在哪限制了，不过能播放网易vip歌曲也够了😔

**Comment by 1715173329** [comment]
实话说如果只是达成这种效果，不如直接用 pyncmd 音源。。

**Review by pan93412 — APPROVED** [review]

**Pull request merged**